src/lib/classes/access-control-list.js

```diff
@@ -0,0 +1,333 @@
+import { namedNode } from '@rdfjs/data-model';
+import solid from 'solid-auth-client';
+import N3 from 'n3';
+import { isEqual } from 'lodash';
+import ldflex from '@solid/query-ldflex';
+
+const PERMISSIONS = {
+  APPEND: 'Append',
+  READ: 'Read',
+  WRITE: 'Write',
+  CONTROL: 'Control'
+};
+
+type Permissions = {
+  agents: null | String | Array,
+  modes: Array<String>
+};
+
+const ACL_PREFIXES = {
+  acl: 'http://www.w3.org/ns/auth/acl#',
+  foaf: 'http://xmlns.com/foaf/0.1/',
+  n: 'http://www.w3.org/2006/vcard/ns#',
+  a: 'http://www.w3.org/ns/auth/acl#type'
+};
+
+class AccessControlList {
+  constructor(owner, documentUri) {
+    this.owner = owner;
+    this.documentUri = documentUri;
+    this.aclUri = `${this.documentUri}.acl`;
+    this.acl = null;
+  }
+
+  set setAcl(acl) {
+    this.acl = acl;
+  }
+
+  static get MODES() {
+    return PERMISSIONS;
+  }
+
+  /**
+   * @function createsQuad Creates a simple quad object
+   * @param {String} subject Subject of the quad
+   * @param {String} predicate Predicate of the quad
+   * @param {String} object Object of the quad
+   */
+  createQuad = (subject, predicate, object) => ({
+    subject,
+    predicate,
+    object
+  });
+
+  /**
+   * @function createQuadList Creates a N3 quad list to later be parsed as Turtle
+   * @param {Array<String>} modes Array of modes
+   * @param {Array<String> | null} agents Array of webId or null if for everyone
+   */
+  createQuadList = (modes: Array<String>, agents: Array<String> | null) => {
+    const { acl, foaf, a } = ACL_PREFIXES;
+    const subject = `${this.aclUri}#${modes.join('')}`;
+    const originalPredicates = [
+      this.createQuad(subject, `${a}`, namedNode(`${acl}Authorization`)),
+      this.createQuad(subject, `${acl}accessTo`, namedNode(this.documentUri))
+    ];
+    let predicates = [];
+    if (agents) {
+      const agentsArray = Array.isArray(agents) ? agents : [agents];
+      const agentsQuads = agentsArray.map(agent =>
+        this.createQuad(subject, `${acl}agent`, namedNode(agent))
+      );
+      predicates = [...originalPredicates, ...agentsQuads];
+    } else {
+      const publicQuad = this.createQuad(subject, `${acl}agentClass`, namedNode(`${foaf}Agent`));
+      predicates = [...originalPredicates, publicQuad];
+    }
+
+    const quadList = modes.reduce(
+      (array, mode) => [
+        ...array,
+        this.createQuad(subject, `${acl}mode`, namedNode(`${acl}${mode}`))
+      ],
+      predicates
+    );
+
+    return quadList;
+  };
+
+  /**
+   * @function createPermissionsTurtle Creates a turtle looking string with specific permissions
```

Comment:
What does "a turtle looking string" mean?

Reply:
Technically it returns a String, but in a turtle format

Reply:
We updated to: "Creates a turtle with specific permissions"